jep/200/README.adoc

```diff
@@ -99,19 +99,22 @@ in favor of the more general `ClassFilter.setDefault`, as is the system property
 
 The new filter has several levels of rules:
 
-* A `CustomClassFilter` implementation may override any behavior. This is discussed in the compatibility section.
+* A custom `ClassFilter` implementation may override any behavior. This is discussed in the compatibility section.
 * Classes mentioned in the original blacklist continue to be rejected.
 * Java arrays and enums are accepted, as these are special formats.
 * Java `Throwable` types are accepted, as they are commonly used in Remoting responses and it is impractical to enumerate every exception that might be thrown.
-* A class defined in Jenkins core, Remoting, or a Jenkins plugin is accepted. This does _not_ apply to classes defined in other JARs packaged in `WEB-INF/lib/*.jar`.
+* A class defined in Jenkins core, Remoting, or a Jenkins module/plugin is accepted. This does _not_ apply to classes defined in other JARs packaged in `WEB-INF/lib/*.jar`.
 * Some classes defined in test utilities, by mocking frameworks, etc., are accepted when running inside tests.
 * Classes defined in a `RemoteClassLoader` (so, already sent to this JVM from a presumably trusted source) are accepted.
   This is needed at least by CloudBees Jenkins Operations Center, and clients of the `master-to-master-api` plugin if there are any.
 * Any other class is accepted if it is explicitly mentioned in a list `whitelisted-classes.txt` bundled in Jenkins core. These consist of:
-** Basic Java utility types (roughly half).
+//oleg-nenashev: What does block us from allowing all types?
+** Basic Java utility types (roughly half, see the reference implementation).
 ** Some other collection types defined in Google Guava.
 ** A handful of safe types defined in some libraries used by Jenkins core.
+//oleg-nenashev: All detached plugins and modules have been checked, right?
 ** A handful of types used by some common plugins known to otherwise fail: Git, Pipeline, Maven, Job DSL.
+*  New APIs for `ClassFilter` management are blacklisted in the Script Security plugin
```

Comment:
Huh? Why? The `script-security` blacklist is for innocent-_looking_ APIs which people have been observed to actually use, typically in Pipeline scripts. It would be ridiculous to even attempt to add every API call in Jenkins which is dangerous for some reason.

Reply:
I think it could be a good practice for new APIs, but it is barely doable without a `@Blacklisted` annotation available in core components

Reply:
There are few Jenkins-specific APIs in the blacklist—only those which we have observed people trying to use innocently, which is very improbable for this obscure API that is clearly linked to security.